worker/pkg/benthos/transformers/generate_ip_address.go

```diff
@@ -0,0 +1,204 @@
+		maxLength, err := args.GetInt64("max_length")
+		if err != nil {
+			return nil, err
+		}
+		ipType, err := args.GetString("ip_type")
+		if err != nil {
+			return nil, err
+		}
+
+		seedArg, err := args.GetOptionalInt64("seed")
+		if err != nil {
+			return nil, err
+		}
+
+		seed, err := transformer_utils.GetSeedOrDefault(seedArg)
+		if err != nil {
+			return nil, err
+		}
+
+		randomizer := rng.New(seed)
+
+		return func() (any, error) {
+			return generateIpAddress(randomizer, IpType(ipType), maxLength)
+		}, nil
+	})
+
+	if err != nil {
+		panic(err)
+	}
+}
+
+func NewGenerateIpAddressOptsFromConfig(config *mgmtv1alpha1.GenerateIpAddress, maxlength *int64) (*GenerateIpAddressOpts, error) {
+	if config == nil {
+		return NewGenerateIpAddressOpts(nil, nil, nil)
+	}
+
+	var ipType *string
+
+	defaultIpType := string(IpV4_Public)
+	ipType = &defaultIpType
+	if config.IpType != nil {
+		v := config.IpType.String()
+		ipType = &v
+	}
+
+	return NewGenerateIpAddressOpts(maxlength, ipType, nil)
+}
+
+func (t *GenerateIpAddress) Generate(opts any) (any, error) {
+	parsedOpts, ok := opts.(*GenerateIpAddressOpts)
+	if !ok {
+		return nil, fmt.Errorf("invalid parsed opts: %T", opts)
+	}
+	return generateIpAddress(parsedOpts.randomizer, IpType(parsedOpts.ipType), parsedOpts.maxLength)
+}
+
+func generateIpAddress(randomizer rng.Rand, ipType IpType, maxLength int64) (string, error) {
+	var ip string
+	var err error
+
+	if ipType == IpV4_V6 {
+		ip = generateIPv6Address(randomizer)
+	} else {
+		ip, err = generateIPv4Address(randomizer, ipType)
+	}
+
+	if err != nil {
+		return "", err
+	}
+
+	if maxLength <= 0 {
+		return ip, nil
+	}
+
+	if int64(len(ip)) > maxLength {
+		return ip[:maxLength], nil
+	}
+
+	return ip, nil
+}
+
+func generateIPv4Address(randomizer rng.Rand, ipType IpType) (string, error) {
+	if ipType == IpV4_Public {
+		return generatePublicIPv4(randomizer)
+	}
+
+	ipRange, exists := ipv4Ranges[ipType]
+	if !exists {
+		return "", fmt.Errorf("unsupported IPv4 type: %s", ipType)
+	}
+
+	return generateIPInRange(randomizer, ipRange.start, ipRange.end), nil
+}
+
+func generateIPv6Address(randomizer rng.Rand) string {
+	groups := make([]string, 8)
+	for i := 0; i < 8; i++ {
+		groups[i] = fmt.Sprintf("%04x", randomizer.Intn(65536))
+	}
+	return strings.Join(groups, ":")
+}
+
+func generatePublicIPv4(randomizer rng.Rand) (string, error) {
+	maxAttempts := 1000
+	for i := 0; i < maxAttempts; i++ {
+		ip := make(net.IP, 4)
+		for i := range ip {
+			ip[i] = byte(randomizer.Intn(256))
+		}
+
+		if !isReservedIP(ip) {
+			return ip.String(), nil
+		}
+	}
+	return "", fmt.Errorf("failed to generate public IP after %d attempts", maxAttempts)
+}
+func isReservedIP(ip net.IP) bool {
+	for _, ipRange := range ipv4Ranges {
+		if inRange(ip, ipRange.start, ipRange.end) {
+			return true
+		}
+	}
+	// Check experimental/reserved range (240.0.0.0/4)
+	experimentalStart := net.ParseIP("240.0.0.0")
+	experimentalEnd := net.ParseIP("255.255.255.255")
```

Comment:
Could move these to be module level to save having to parse the IP string every time.